### PR TITLE
ZENKO-3283: CRUD acccess keys

### PR DESCRIPTION
**Pull request by theocerutti**
**No Keys:**
image

**Keys:**
Note: Backend is not ready for LastUsed column.
image

> If vertical overflow > scroll allowed on full tab level (“properties”)

image

### CODE REVIEW

File: src/js/IAMClient.js
Comment (nicolas2bert) [inline src/js/IAMClient.js]: Let's get endpoints from config.

File: src/react/account/details/properties/AccountKeys.jsx
Comment (nicolas2bert) [inline src/react/account/details/properties/AccountKeys.jsx]: If you want to retrieve account access keys, do not specify the username parameter. 
`If the UserName field is not specified, the user name is determined implicitly based on the AWS access key ID used to sign the request.`
You will have to retrieve the temporary access key of this specific account (using assumeRoleWithWebIdentity) to make the list access keys call.

File: src/react/account/details/properties/AccountKeys.jsx
Comment (nicolas2bert) [inline src/react/account/details/properties/AccountKeys.jsx]: To create account access key, use `API.generateKeyConfigurationOverlayUser()`
You will find its definition here: http://management.zenko.local/swagger.json

File: src/react/actions/sts.js
Comment (nicolas2bert) [inline src/react/actions/sts.js]: We might want to call the method `login` to be consistant.

File: src/js/IAMClient.js
Comment (theocerutti) [reply src/js/IAMClient.js]: done

File: src/react/actions/sts.js
Comment (theocerutti) [reply src/react/actions/sts.js]: done

File: src/react/account/details/properties/AccountKeys.jsx
Comment (theocerutti) [reply src/react/account/details/properties/AccountKeys.jsx]: done

File: src/react/account/details/properties/AccountKeys.jsx
Comment (theocerutti) [reply src/react/account/details/properties/AccountKeys.jsx]: This ticket doesn't ask to add the access key creation, so I remove this button "Create Access Key".

From jira:
> The creation form when clicking on the create button will be in a second iteration.

**Comment by Cuervino** [comment]
An issue that I spot with the scrollbar: it should be on the very right of the container, and it should be space between the scrollbar and the button Delete.
image

**Comment by theocerutti** [comment]
I have no idea why tests doesn't pass the CI. (npm run test && npm run test:lint doesn't throw any errors)
Maybe because I modified the webpack configuration ?

File: src/react/account/Accounts.jsx
Comment (nicolas2bert) [inline src/react/account/Accounts.jsx]: `selectAccountID` is a side effects operation. We might want to use React `useEffect` method?
Also, I do not think we want to "switch account" (reloading clients and re-fetching items) every time we select a new account.
Instead, what we can do for `deleteAccountAccessKey` and `listAccountAccessKey` actions method (in actions/account.js):
- retrieve temporary keys with STS.assumeRoleWithWebIdentity (generate role with the right account name)
- instantiate new IAM client  with the temporary keys.
- call the IAM respective method.

Comment (theocerutti) [reply src/react/account/Accounts.jsx]: done

File: src/react/actions/account.js
Comment (nicolas2bert) [inline src/react/actions/account.js:133]: Looks like some code are duplicated here with the function `listAccountAccessKeys`.

File: src/react/actions/sts.js
Comment (nicolas2bert) [inline src/react/actions/sts.js]: Not needed here.

File: src/react/reducers/initialConstants.js
Comment (nicolas2bert) [inline src/react/reducers/initialConstants.js]: not needed here, might be confusing for reviewers.

File: src/react/utils/actions.js
Comment (nicolas2bert) [inline src/react/utils/actions.js]: not needed.

File: src/types/state.js
Comment (nicolas2bert) [inline src/types/state.js]: Not needed.

File: src/react/reducers/index.js
Comment (nicolas2bert) [inline src/react/reducers/index.js]: not needed.

File: src/react/reducers/iam.js
Comment (nicolas2bert) [inline src/react/reducers/iam.js]: not needed.

File: src/react/actions/account.js
Comment (nicolas2bert) [inline src/react/actions/account.js]: Let's keep the stored clients in sync with the account context. Meaning, here let's instantiate a new iamClient and not retrieve/modify the stored one.

File: src/react/actions/__tests__/utils/testUtil.js
Comment (nicolas2bert) [inline src/react/actions/__tests__/utils/testUtil.js]: Not needed.

File: src/js/IAMClient.js
Comment (nicolas2bert) [inline src/js/IAMClient.js]: not used.

File: src/react/account/details/properties/AccountKeys.jsx
Comment (nicolas2bert) [inline src/react/account/details/properties/AccountKeys.jsx]: `accessKeyList` should be retrieved from the `account`  "store table" (please create one if it does not exist) instead of `user`, to avoid confusion.

**Review by nicolas2bert — COMMENTED** [review]
First iteration, let's clean things up to ease the review.

File: src/react/account/details/properties/AccountProperties.jsx
Comment (nicolas2bert) [inline src/react/account/details/properties/AccountProperties.jsx:26]: We might want to move the account keys to its own **Keys** "account tab". I remember creating "Keys" component in `account/details`. 
That change should be quite easy and has two main advantages
- Separation of concern.
- We would have more space to display items.
- No need to load access keys on every account selection anymore.

Waiting for @terrydervaux approval

Comment (terrydervaux) [reply src/react/account/details/properties/AccountProperties.jsx:26]: Long story short, I want to keep root user access keys in the `Properties` tab.

We planned to have 7 tabs in the account page design. We want to use our "separation of concern capabilities" for account consumption metrics, storage locations, and IAM entities.

It's not a [best practice to have, keep, and use root user access keys](https://docs.aws.amazon.com/general/latest/gr/aws-security-audit-guide.html#aws-security-audit-review-account), our client should not have "a lot of keys". The table will be most of the time empty.

Tabs planned for account page:
- Metrics
- Users
- Groups
- Roles 
- Policies
- Storage location
- Properties

Comment (nicolas2bert) [reply src/react/account/details/properties/AccountProperties.jsx:26]: In my opinion, most of these items should have their own page:
- Metrics
- Users
- Groups
- Roles
- Policies
- Storage location
- Properties

File: src/react/account/details/properties/AccountKeys.jsx
Comment (terrydervaux) [inline src/react/account/details/properties/AccountKeys.jsx]:
```suggestion
            <h3>Root user access keys details</h3>
```

File: src/react/account/details/properties/AccountKeys.jsx
Comment (terrydervaux) [inline src/react/account/details/properties/AccountKeys.jsx]:
```suggestion
                        Security Status: root user access keys gives unrestricted
                        access to account resources. It is recommended to delete
                        root access keys and use IAM user access keys.
```
@wabernatScality fell free to suggest a better wording.

File: src/react/account/details/properties/AccountKeys.jsx
Comment (terrydervaux) [inline src/react/account/details/properties/AccountKeys.jsx]:
```suggestion
            Header: 'Access Key ID',
```

**Review by terrydervaux — APPROVED** [review]
Feature-wise LGTM.

@Cuervino please, review the UI design.

File: src/react/account/details/properties/AccountKeys.jsx
Comment (wabernat) [reply src/react/account/details/properties/AccountKeys.jsx]:
```suggestion
                        Security Status: Root user access keys give unrestricted
                        access to account resources. It is a best practice to delete
                        root access keys and use IAM user access keys instead.
```

**Review by nicolas2bert — APPROVED** [review]
Tests for `deleteAccountAccessKey` and `deleteAccountAccessKey` Redux actions would be addressed in a following PR.

**Comment by theocerutti** [comment]
@bert-e approve

**Pull request merged**
